docs/spec/spec.md

```diff
@@ -0,0 +1,1090 @@
+    int messageCount = -1;
+    # Array of Azure service bus message representation (Array of Message records)
+    Message[] messages = [];
+|};
+```
+
+## 3. Message sender
+
+A message sender is responsible for sending Azure service bus messages to a queue or topic on the Azure service bus instance.
+
+### 3.1. Configurations
+
+- `EntityType` enum represents the Azure service bus resource entity type.
+
+```ballerina
+public enum EntityType {
+    QUEUE = "queue",
+    TOPIC = "topic"
+}
+```
+
+- `AmqpRetryOptions` represents options that can be used to specify the underlying client retry behavior.
+
+```ballerina
+public type AmqpRetryOptions record {|
+    # Maximum number of retry attempts
+    int maxRetries = 3;
+    # Delay between retry attempts in seconds
+    decimal delay = 10;
+    # Maximum permissible delay between retry attempts in seconds
+    decimal maxDelay = 60;
+    # Maximum duration to wait for completion of a single attempt in seconds
+    decimal tryTimeout = 60;
+    # Approach to use for calculating retry delays
+    AmqpRetryMode retryMode = asb:FIXED;
+|};
+
+# Represents the type of approach to apply when calculating the delay between retry attempts.
+public enum AmqpRetryMode {
+    # Retry attempts happen at fixed intervals; each delay is a consistent duration.
+    FIXED,
+    # Retry attempts will delay based on a backoff strategy, where each attempt will increase the duration that it waits before retrying.
+    EXPONENTIAL
+};
+```
+
+- `ASBServiceSenderConfig` record represents the Azure service bus message sender configurations.
+
+```ballerina
+public type ASBServiceSenderConfig record {
+    # An enumeration value of type EntityType, which specifies whether the connection is for a topic or a queue. 
+    # The valid values are TOPIC and QUEUE
+    EntityType entityType;
+    # A string field that holds the name of the topic or queue
+    string topicOrQueueName;
+    # A string field that holds the Service Bus connection string with Shared Access Signatures
+    string connectionString;
+    # Retry configurations related to underlying AMQP message sender
+    AmqpRetryOptions amqpRetryOptions = {};
+};
+```
+
+### 3.2. Initialization
+
+- The `asb:MessageSender` can be initialized by providing the `asb:ASBServiceSenderConfig`.
+
+```ballerina
+# Initializes an Azure service bus message sender.
+# ```
+# configurable string connectionString = ?;
+# asb:ASBServiceSenderConfig senderConfig = {
+#   connectionString: connectionString,
+#   entityType: asb:QUEUE,
+#   topicOrQueueName: "testQueue1" 
+# };
+# asb:MessageSender sender = check new(senderConfig);
+# ```
+#
+# + config - Azure service bus sender configuration
+# + return - The `asb:MessageSender` or an `asb:Error` if the initialization failed
+public isolated function init(asb:ASBServiceSenderConfig config) returns asb:Error?;
+```
+
+### 3.3. Functions
+
+- To send a message to a queue or topic, the `send` function can be used.
```

Comment:
Probably we can categorize the functions into H4 headers (and include in the ToC). WDYT?

Reply:
In the current organization of the spec we have following hierarchy.

```
H2 x. Client
    H3 x.1 Configurations
    H3 x.2 Initialization
    H3 x.3 Functions
```

If we move the `Functions` section into H4 I think we need to move it as a sub-category of the previous section, which in this case `Initialization` section. IMO it would confusing to the developers who refer to the spec.

Reply:
As discussed offline we will not do the proposed restructure to maintain consistency among the other library module specifications.